src/app/pages/single-course-page/CoursePage.tsx

```diff
@@ -0,0 +1,174 @@
+    (state) => state.persistedReducer.coursesListPage
+  );
+  const { coursesProgress } = useAppSelector(
+    (state) => state.persistedReducer.coursePage
+  );
+  const { setToken } = coursesListPageSlice.actions;
+  const { setCourseProgress, setLessonProgress } = coursePageSlice.actions;
+  const dispatch = useDispatch();
+
+  const client = new Client();
+  const navigate = useNavigate();
+  const [courseData, setCourseData] = useState({} as IGetCourse);
+
+  useEffect(() => {
+    async function getData() {
+      async function getToken() {
+        try {
+          const responseData = (await client.auth.getToken()).data;
+          dispatch(setToken(responseData.token));
+          return responseData.token;
+        } catch (error) {
+          console.log(error);
+          return;
+        }
+      }
+
+      async function getCourse(token: string) {
+        try {
+          const responseData = (await client.courses.getCourse(courseId, token))
+            .data;
+          const lessons = responseData.lessons.sort(
+            (a, b) => a.order - b.order
+          );
+          responseData.lessons = lessons.slice();
+          console.log(responseData);
+          setCourseData(responseData);
+        } catch (error) {
+          console.log(error);
+          if (error instanceof AxiosError) {
+            if (error.response?.status === 401) {
+              return authErrorResponse;
+            } else {
+              navigate("/error/");
+            }
+          }
+        }
+      }
+
+      const response = await getCourse(token);
+      if (response === authErrorResponse) {
+        const newToken = await getToken();
+        await getCourse(newToken);
+      }
+    }
+
+    getData();
+  }, []);
+
+  useEffect(() => {
+    if (courseData.lessons) {
+      if (!coursesProgress[courseId]) {
+        const unlockedLessons = courseData.lessons.filter(
+          (lesson) => lesson.status === unlockedStatus
+        );
+        const courseProgress = unlockedLessons.reduce(
+          (arr, lesson) => ({ ...arr, [lesson.id]: 0 }),
+          { preview: 0 }
+        );
+        dispatch(setCourseProgress({ courseId, courseProgress }));
+      }
+    }
+
+    var video = document.getElementById("previewVideo");
+    if (Hls.isSupported() && video instanceof HTMLMediaElement) {
+      console.log(coursesProgress);
+      var hls = new Hls();
+      hls.loadSource(courseData.meta.courseVideoPreview.link);
+      hls.attachMedia(video);
+      video.currentTime = coursesProgress[courseId]
+        ? coursesProgress[courseId]["preview"]
+        : 0;
+
+      video.addEventListener("timeupdate", function () {
+        dispatch(
+          setLessonProgress({
+            courseId,
+            lessonId: "preview",
+            lessonProgressTime: this.currentTime,
+          })
+        );
+      });
+    }
+  }, [courseData]);
+
+  return (
+    <>
+      {Object.keys(courseData).length !== 0 ? (
```

Comment:
There is no need to use an additional Fragment here

```suggestion
  return (
    Object.keys(courseData).length !== 0 ? (
```

Reply:
Thanks! Done.